### DISCUSSION TITLE
ESP8266-MicroPython install - No files

### DISCUSSION THREAD

**Discussion by salexander319** [micropython/micropython#15810] [Hardware & Peripherals] [answered]
I have installed MicroPython on an ESP8266-01 with 1 MiB of flash. I noticed there were no files on this configuration, and I couldn't install any. Then I saw that there is no FAT filesystem support for this device. Thus, it would seem that there is no way to actually use this device in a stand-alone situation with MicroPython since there is no main.py to run on power-up. Is this the case or am I missing some trick?

**Comment by davefes** [comment]
Do you get a REPL prompt in whatever terminal program you are using?

How did you load the firmware?

**Comment by glenn20** [comment]
That sounds odd. Did you erase the flash on the device beforeflashing the new micropython firmware? If the flash was not erased first, micropython will not initialise the filesystem on first boot. See https://docs.micropython.org/en/latest/esp8266/tutorial/intro.html#deploying-the-firmware.

**Comment by robert-hh** [comment]
Did you flash the 1MB version of the firmware? It is at the bottom of the ESP8266 download page https://micropython.org/download/ESP8266_GENERIC/

**Comment by salexander319** [comment]
On the Python download page I noted the comment below after struggling with how to save a main.py. I used ESPTOOL to erase the flash and then flash the 1MB download. I was quite excited when I got a python prompt in my Thonny terminal. I could enter and execute python code manually. I could also run scripts I'm editing in Thonny. But there were no files shown in the Micropython device panel. And if I try to save to main.py on the device it gives me an error: OSError: Could not open file /main.py for writing, output.  So, it would seem with no file system support it makes sense that there are no files. But I'm not sure what the point of the 1MB release is if I can't have it run a program on its own.

> For boards with 1MiB or 512kiB of flash, two variants are provided with reduced functionality. The 1MiB variant removes asyncio and FAT-filesystem support as well as some modules from micropython-lib. The 512kiB variant further removes all filesystem support, as well as framebuffer support, some Python language features, and has less detailed error messages.

**Reply by davefes** [reply]
I wonder if `no FAT filesystem support` doesn't mean that there is `no FAT filesystem`.  It has been a few years since I worked with the ESP8266.

Just my advise ... if you are starting out wanting to use ESP for WiFi-connected projects I would suggest switching to the ESP32.

**Reply by glenn20** [reply]
No FAT-fs support means that the default micropython firmware only supports LittleFS filesystems (which are generally regarded as better for use on flash storage than FAT fs).

I have never used Thonny, so can't help with that.

PS. I use [mpremote](https://docs.micropython.org/en/latest/reference/mpremote.html) to copy files to the device flash storage. You could try that if you want to see if it is a micropython or thonny issue.

**Reply by robert-hh** [reply, marked as answer]
I just checked with a 1M ESP8266. There is a small LFS2 file system. You should be able to verify with:

```
import os
os.statvfs("")
```
Here, it returns `(4096, 4096, 98, 93, 93, 0, 0, 0, 0, 255)`, telling that there are 98 blocks of 4k size available.

You will not see the board's file system in the file manager of the PC. I can use Thonny to access the file system setting the interface properly to /dev/ttyUSB0. But I normally use mpremote as less intrusive tool. If you prefer to use a IDE, look at ViperIDE, which also has no unwanted side effects.

**Reply by salexander319** [reply]
I started from scratch and made sure I had the correct bin file and now I'm seeing what I should. I see the correct return from the os.statvfs command as shown in robert-hh's post. I must have flashed the wrong file. Thonny is working fine. Thank everyone for the comments and help.
